### ISSUE TITLE
Logs view implementation for Visualization

### ISSUE THREAD

**Issue by nidhisinghai** [opensearch-project/observability#773]
Hello @rinku-kumar-psl

This Issue requires a below Implementation -
 Logs view implementation under Visualization and its config options implementation as per issue 697.

@abasatwar @spattnaik

**Comment by ruchika-narang** [comment]
Hey @mengweieric @btzeng 

We have started implementation on this ticket and have few queries related to it. 

1. What data should we show in logs visualizations. Eg: If a user queries with multiple fields, then do we have to show the raw response with all the fields or the filtered response with field specified by users.
2. If filtered PPL response doesn't have any field of timestamp, do we have to disable the time from config panel or we pick up the timestamp from raw data received?
3. We see unique labels and common labels in the config panel. What are these keys in the response data?
4. In config panel, under the title Deduplication, we have options as (None, exact, numbers, signature). Could you please let us know what do these options signify.
5. Also do we have to show the log view visualization for every query?
6. When we open the accordion, we see an ad-hoc statistics icon beside every field in the accordion, do we have to implement that too?

Thanks,
Ruchika
CC: @anirudha @abasatwar @spattnaik

**Comment by mengweieric** [comment]
Hi @ruchika-narang,

What we should have for log config panel would be

*Panel options*

* title
* descriptions

Logs

*Time (switch)*
Show or hide the time column. This is the timestamp associated with the log line as reported from the data source. Disabled if index does not contain timestamp

*Font size (number only input)*
Changes the size of the log font

*Order (filter buttons)*
Time asc / desc

1. We reuses the log view from event tab.
2. Yes, we should disable any timestamp related config option.
5. Could share a little more about what do you mean for every query?

For 2,3,4 and 6, for now we just need above config items to be implemented. I'll share the doc with more details for this part once we finished doc review and iterations.

**Comment by shankha-das** [comment]
Hi @mengweieric , @btzeng
Please refer attached video for implementation we have done

https://user-images.githubusercontent.com/93648901/177711276-f5c28eec-9c2d-4854-adc8-1bfb19e5d914.mp4

Please share your inputs

Thanks

**Comment by pratibhapandey16** [comment]
Hi @shankha-das,
Please find the below observations.
1. On clicking reset button the Label size value is not changing to blank.
2. Details are not getting updated accordingly when we select negative value for the Label Size.

**Comment by shankha-das** [comment]
Hi @btzeng ,

As discussed in the Demo call, we had concluded to not let user select Prettify JSON when Wrap Lines is enabled. As default, wrap lines and prettify JSON views are completely different and not related to each other, I have combined them to show as a button group of these three options from which the user can select one and it is more user friendly. Following is the screenshot of the implementation, let me know if this looks good to you and I can proceed with this.

![MicrosoftTeams-image (12)](https://user-images.githubusercontent.com/93648901/178456414-35b60647-5972-4714-a59d-e5a2b2f703a1.png)

cc: @anirudha @mengweieric @spattnaik @abasatwar
